apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Edit/EditBusinessRules.tsx

```diff
@@ -0,0 +1,274 @@
+            });
+        });
+    }, [ruleId]);
+
+    const findSourceQuestion = (questionIdentifier?: string): PagesQuestion | undefined => {
+        let result: PagesQuestion | undefined = undefined;
+        page?.tabs?.map((tab: PagesTab) => {
+            tab.sections?.map((section: PagesSection) => {
+                section.subSections?.map((subsection: PagesSubSection) => {
+                    subsection.questions?.map((question: PagesQuestion) => {
+                        if (question.question === questionIdentifier) {
+                            result = question;
+                        }
+                    });
+                });
+            });
+        });
+        return result;
+    };
+
+    const findTargetQuestion = (targets?: string[]): PagesQuestion[] => {
+        const targetQuestions: PagesQuestion[] = [];
+        page?.tabs?.map((tab: PagesTab) => {
+            tab.sections?.map((section: PagesSection) => {
+                section.subSections?.map((subsection: PagesSubSection) => {
+                    subsection.questions?.map((question: PagesQuestion) => {
+                        targets?.map((target) => {
+                            if (target === question.question) {
+                                targetQuestions.push(question);
+                            }
+                        });
+                    });
+                });
+            });
+        });
+        return targetQuestions;
+    };
+
+    const findTargetSubsection = (targets?: string[]): PagesSubSection[] => {
+        const targetQuestions: PagesSubSection[] = [];
+        page?.tabs?.map((tab: PagesTab) => {
+            tab.sections?.map((section: PagesSection) => {
+                section.subSections?.map((subsection: PagesSubSection) => {
+                    targets?.map((target) => {
+                        if (target === subsection.questionIdentifier) {
+                            targetQuestions.push(subsection);
+                        }
+                    });
+                });
+            });
+        });
+        return targetQuestions;
+    };
+
+    const fetchSourceValues = (valueSet?: string) => {
+        if (valueSet) {
+            fetch(valueSet);
+        }
+    };
+
+    useEffect(() => {
+        if (selectedSourceValues && options) {
+            const matchedValues = options.filter((opt) => selectedSourceValues?.find((val) => val === opt.name));
+            const newValues = matchedValues.map((value) => ({ id: value.value, text: value.name }));
+            form.setValue('sourceValues', newValues);
+        }
+    }, [JSON.stringify(selectedSourceValues), JSON.stringify(options)]);
+
+    const redirectToLibrary = () => {
+        if (page?.id) {
+            navigate(`/page-builder/pages/${page.id}/business-rules`);
+        } else {
+            navigate(`../`);
+        }
+    };
+
+    const onSubmit = form.handleSubmit(async (data) => {
+        try {
```

Comment:
small nitpick, in some places we have .then(), in others we have  async await with try/catch, it would be nice to be consistent, preferably just everything async await